chain/transaction.go

```diff
@@ -162,6 +163,8 @@ func (t *Transaction) Size() int { return t.size }
 
 func (t *Transaction) ID() ids.ID { return t.id }
 
+func (t *Transaction) SetID(id ids.ID) { t.id = id }
```

Comment:
We should avoid exporting a function that overrides the ID of the transaction since it should always match the contents of the transaction